packages/preferences/src/browser/abstract-resource-preference-provider.ts

```diff
@@ -12,7 +12,7 @@ import { ILogger, DisposableCollection, Resource, Event, Emitter, ResourceProvid
 import { PreferenceProvider } from '@theia/core/lib/browser/preferences';
 
 @injectable()
-export abstract class AbstractResourcePreferenceProvider implements PreferenceProvider {
+export abstract class AbstractResourcePreferenceProvider extends PreferenceProvider {
```

Comment:
`onDidPreferencesChangedEmitter`, `toDispose` and `dispose` can be removed, since they are provided by the super class

Reply:
done